src/ml/neural_net/model_spec.hpp

```diff
@@ -305,6 +305,17 @@ class model_spec {
   void add_exp(const std::string& name, const std::string& input);
 
 
+  /**
+   * Appends a layer that scales an entire batch with a single value
+   *
+   * \param name The name of the layer and its output
+   * \param input The name of the layer's input
+   * \param shape_c The shape of the channel
+   * \param weight_initializer_fn Callback used to initialize the weights
+   */
+  void add_scale(const std::string& name, const std::string& input,
+                 weight_initializer scale_initializer_fn);
```

Comment:
Could we instead just generalize the existing `add_scale` to take an arbitrary shape vector?

Reply:
Just have it take `const std::vector<size_t>& scale_shape` or something like that?